- adding v4 .Net lambda sample project

sbhuller opened the pull request.

sbhuller committed:
- adding v4 .Net lambda sample project

ckissinger commented on RightslineSampleLambdaDotNetV4/RightslineSampleLambdaDotNetV4/Function.cs:
```diff
@@ -0,0 +1,139 @@
+                        Console.WriteLine($"Cannot find Catalog Items with template id {messageEntity.Entity.Template.TemplateId}.");
+                        Console.WriteLine("Lambda processing aborted");
+                    }
+
```
maybe add an example of grabbing additional information from a rightset in here also? or getting all catalog items associated to a rightset when it is changed?

ckissinger commented on RightslineSampleLambdaDotNetV4/RightslineSampleLambdaDotNetV4/Models/EntityRelationshipModel.cs:
```diff
@@ -0,0 +1,44 @@
+    public class EntityRelationshipModel
+    {
+        public int? Id { get; set; }
+        public string ParentURL { get; set; }
```
I think this is the v3 model, in v4 we moved away from these URL string properties and we also added the template and status of the parent and child

adammorton9 commented on RightslineSampleLambdaDotNetV4/RightslineSampleLambdaDotNetV4/Enumerations/Enumerations.cs:
```diff
@@ -0,0 +1,33 @@
+        [Description("catalog-item")]
+        CatalogItem = 1,
+        [Description("rightset")]
+        Rightset = 3,
```
Use "right" instead of "rightset"

sbhuller replied:
Resolved in branch.

sbhuller committed:
- Rename from rightset to right
- Add additional constants for attribute indexing.

sbhuller committed:
- updated models to v4 from RL project
- added additional snippet for fetching right information
- added additional snippet for fetching catalog items associated with rightset on a right updated message
